graalpy-install-package/src/main/java/com/oracle/example/graalpy/Main.java

```diff
@@ -62,40 +63,77 @@ public static void main(String[] args) throws Exception {
+      .option("python.AlwaysRunExcepthook", "true")
+      // Force to automatically import site.py module, to make Python packages available
+      .option("python.ForceImportSite", "true")
+      // The sys.executable path, a virtual path that is used by the interpreter to discover packages
+      .option("python.Executable", vfs.resourcePathToPlatformPath(VENV_PREFIX) + (VirtualFileSystem.isWindows() ? "\\Scripts$\\python.exe" : "/bin/python"))
+      // Set the python home to be read from the embedded resources
+      .option("python.PythonHome", vfs.resourcePathToPlatformPath(HOME_PREFIX))
+      // Do not warn if running without JIT. This can be desirable for short running scripts
+      // to reduce memory footprint.
+      .option("engine.WarnInterpreterOnly", "false")
+      // Set python path to point to sources stored in src/main/resources/vfs/proj
+      .option("python.PythonPath", vfs.resourcePathToPlatformPath(PROJ_PREFIX))
+      .build();
+    return context;
+  }
 
   /**
    * This creates an instance of the PyfigletWrapper type and returns it
    * mapped to the {@link PyfigletProxy} interface.
    */
   static PyfigletProxy createPyfigletProxy() {
-    Context context = Context.newBuilder(PYTHON).
-    // It is a good idea to start with allowAllAccess(true) and only when everything
-    // is working to start trying to reduce it.
-    // See the GraalVM docs for fine-grained permissions.
-        allowAllAccess(true).
-        // Python virtualenvs work by setting up their initial package paths based on
-        // the runtime path of the python executable.
-        // Since we are not executing from the python executable,
-        // we need to set this option to what it would be
-        option("python.Executable", VENV_EXECUTABLE).
-        // The actual package setup only happens inside Python's "site" module. 
-        // This module is automatically imported when starting the Python executable,
-        // but there is an option to turn this off even for the executable.
-        // To avoid accidental file system access,
-        // we do not import this module by default.
-        // Setting this option to true after setting the python.Executable option
-        // ensures we import the site module at startup, but only within the virtualenv.
-        option("python.ForceImportSite", "true").build();
-    InputStreamReader code = new InputStreamReader(Main.class.getClassLoader().getResourceAsStream(SOURCE_FILE_NAME));
+    Context context = getContext();
+    // Import and evaluate the Python file we provide in the resources directory.
+    // Holding on to the source object is recommended if we plan to create multiple
+    // contexts with the same polyglot engine to share JIT compiled code.
     Source source;
     try {
-      source = Source.newBuilder(PYTHON, code, SOURCE_FILE_NAME).build();
+      source = Source.newBuilder(PYTHON, "import pyfigletwrapper", "<internal>").build();
```

Comment:
This is fine, just wanted to mention that you could've also done this:

```suggestion
      source = Source.newBuilder(PYTHON, "import pyfigletwrapper; PyfigletWrapper", "<internal>").build();
```

Then, `context.eval(source);` should return the class. Sharing via the polyglot binding is still good to show.

Reply:
I think you should not use `<internal>` here, maybe something like this:
```suggestion
      source = Source.newBuilder(PYTHON, "import pyfigletwrapper", "pyfigletwrapper").build();
```